bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptExtensionProvider.java

```diff
@@ -15,46 +15,51 @@
 import java.util.Collection;
 import java.util.Map;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * A {@link ScriptExtensionProvider} can provide variable and types on ScriptEngine instance basis.
  *
  * @author Simon Merschjohann - Initial contribution
  */
+@NonNullByDefault
 public interface ScriptExtensionProvider {
 
     /**
      * These presets will always get injected into the ScriptEngine on instance creation.
      *
      * @return collection of presets
      */
-    public Collection<String> getDefaultPresets();
+    Collection<String> getDefaultPresets();
 
     /**
      * Returns the provided Presets which are supported by this ScriptExtensionProvider.
      * Presets define imports which will be injected into the ScriptEngine if called by "importPreset".
      *
      * @return provided presets
      */
-    public Collection<String> getPresets();
+    Collection<String> getPresets();
 
     /**
      * Returns the supported types which can be received by the given ScriptExtensionProvider
      *
      * @return provided types
      */
-    public Collection<String> getTypes();
+    Collection<String> getTypes();
 
     /**
      * This method should return an Object of the given type. Note: get can be called multiple times in the scripts use
      * caching where appropriate.
      *
      * @param scriptIdentifier the identifier of the script that requests the given type
      * @param type the type that is requested (must be part of the collection returned by the {@code #getTypes()} method
-     * @return the requested type (non-null)
+     * @return the requested type or null
```

Comment:
I am not sure if this is a valid change but as all existing implementations do not take care about throwing IAEs I changed it ...

Reply:
The changes makes sense to me.